File: interview_code/15_number_of_1_in_binary/c/number_of_1_in_binary.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>








int count_1_in_number_each(int number)
{
	int count = 0;
	int flag = 1;
	
	while(flag){
		if(number & flag){
			count++;
		}
		flag <<= 1;
	}
	
	return count;
}



int count_1_in_number_cur(int number)
{
	int count = 0;
	
	while(number){
		number = (number-1)&number;
		count++;
	}
	return count;
}








void main(void)
{
	int number = 100;
	
	printf("The number %d have %d '1' in binary\n", number, count_1_in_number_each(number));
	printf("The number %d have %d '1' in binary\n", number, count_1_in_number_cur(number));
}

File: interview_code/15_number_of_1_in_binary/python/number_of_1_in_binary.py
# -*- coding: utf-8 -*-


'''
def count_1_in_number_each_bit(number):
    flag = 1
    count = 0
    i = 0
    
    while(int(flag)):
        if number & flag:
            count += 1
        flag <<= 1
        i += 1
    return count
'''

def count_1_in_number_cur_bit(number):
    count = 0
    
    while(number):
        number = number&(number-1)
        count += 1
        
    return count



if __name__ == '__main__':
    number = 100
    #print('The number %d have %d "1" in binary' % (number, count_1_in_number_each_bit(number)))
    print('The number %d have %d "1" in binary' % (number, count_1_in_number_cur_bit(number)))